pkg/backup/backup_compaction_test.go

```diff
@@ -315,6 +315,63 @@ crdb_internal.json_to_pb(
 	})
 	// TODO (kev-cao): Once range keys are supported by the compaction
 	// iterator, add tests for dropped tables/indexes.
+	t.Run("automatic compaction after scheduled backup", func(t *testing.T) {
+		defer testutils.HookGlobal(&scheduleOnly, false)()
+		db.Exec(t, "SET CLUSTER SETTING backup.compaction.threshold = 4")
+		db.Exec(t, "SET CLUSTER SETTING backup.compaction.window = 2")
+		db.Exec(t, "CREATE TABLE foo (a INT, b INT)")
+		defer func() {
+			db.Exec(t, "DROP TABLE foo")
+		}()
+		db.Exec(t, "INSERT INTO foo VALUES (1, 1)")
+		db.Exec(t, fmt.Sprintf(fullBackupAostCmd, 8, getTime(t)))
+		db.Exec(t, "INSERT INTO foo VALUES (2, 2)")
+		db.Exec(t, fmt.Sprintf(incBackupCmd, 8))
+		db.Exec(t, "INSERT INTO foo VALUES (3, 3)")
+		db.Exec(t, fmt.Sprintf(incBackupCmd, 8))
```

Comment:
Honestly, working on this again, I'm not a fan of just putting these numbers to indicate a bucket and having so much repetition. I think a `tmpDir` would be better, and we could update our helper functions.

Just a code quality thing though, not urgent but maybe nice to do after the release.